cpp/src/arrow/type_fwd.h

```diff
@@ -629,6 +634,17 @@ std::shared_ptr<Schema> schema(
     std::vector<std::shared_ptr<Field>> fields,
     std::shared_ptr<const KeyValueMetadata> metadata = NULLPTR);
 
+/// \brief Create a Schema instance
+///
+/// \param fields the schema's fields in the form of initializer_list
+/// \param metadata any custom key-value metadata, default null
+/// \return schema shared_ptr to Schema
+ARROW_EXPORT
+std::shared_ptr<Schema> schema(
+    const std::initializer_list<std::pair<std::string, std::shared_ptr<DataType>>>&
```

Comment:
Initializer list parameters should not be declared with `const&`
```suggestion
    std::initializer_list<std::pair<std::string, std::shared_ptr<DataType>>>
```